### PR TITLE
Feature/planet detail

### PR DESCRIPTION
**Pull request by alexthompson207**
# PR PlanetParty
**Connor A-L, Alex T & Katie B**

### What does this do?

- [ ] Fix
- [ X ] Feature
- [ X ] Refactor
- [ ] Style
- [ ] Testing

### Are there any known bugs?

- [ X ] No
- [ ] Yes (if so please disclose in Notes for Reviewer & Next Iterations)

### Summary of the changes
Create a single Planet view component and add basic styling. Also refactor API call to get data from a different API and add functions to clean data.

### What ticket(s) do the changes resolve?
Single planet view and fetch API data. Solves CORS error we were getting with old API.

Please Link Issues to this PR as necessary.

### Notes for the Reviewer
The planet info fact is hard coded so that data will not change when sort by type.

### How to Test Changes?
Pull down branch, make sure state is holding API info. No CORS errors of response errors in the console.

### Next Iterations

- Pass planet fact to each card when correct sort button is clicked
- Allow a user to click on a card, that will then display single planet view with respective facts
- Do we need to make a single API call for when a planet is clicked?

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
+  //       diameter: info.diameter,
+  //       gravity: info.gravity,
+  //       length_of_day: info.length_of_day,
+  //       distance_from_sun: info.distance_from_sun,
+  //       mean_temperature: info.mean_temperature,
+  //       number_of_moons: info.number_of_moons
+  //     }
+  //   }))
+  //   .then(response => response.filter(planet => (planet.name !== 'Moon' && planet.name !== 'Pluto')))
+  return fetch('https://api.le-systeme-solaire.net/rest.php/bodies?data=%20id%2C%20englishName%2C%20moons%2C%20mass%2C%20gravity%2C%20massValue%2C%20massExponent%2C%20moon%2C%20meanRadius%2C%20sideralOrbit%2C%20sideralRotation%2C%20semimajorAxis&filter%5B%5D=isPlanet%2Cneq%2Ctrue&filter%5B%5D=')
```
Comment (knees4bees) [inline src/apiCalls.js:20]: Well well done getting this call to the new API up and running!

File: src/apiCalls.js [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
+      return {
+        id: info.id,
+        name: info.englishName,
+        mass: info.mass.massValue,
+        diameter: (info.meanRadius * 2),
+        gravity: info.gravity,
+        length_of_day: Math.abs(info.sideralRotation),
+        distance_from_sun: info.semimajorAxis,
+        length_of_year: info.sideralOrbit,
+        number_of_moons: info.moons?.length || 0
```
Comment (ConnorAndersonLarson) [inline src/apiCalls.js:35]: What a legend. We'll have to maybe `Math.floor/ceiling` some of this data in the future, but this works great.

File: src/components/Planet/Planet.tsx [hunk 3]
```diff
@@ -1,23 +1,22 @@
...
 }
 
 
-const Planet: React.FC<PlanetData> = ({ id, name, planetFact }): JSX.Element => {
+const Planet: React.FC<PlanetProps> = ({ id, name, planetFact }): JSX.Element => {
 
   return (
     <article className='planet-card' id={id.toString()}>
-      <img className='planet-icon' alt='earth' src={earth}></img>
+      <img className='planet-icon' alt='earth' src={`../planets/${name}.png`} />
```
Comment (ConnorAndersonLarson) [inline src/components/Planet/Planet.tsx:15]: Nice clean image finding.

File: src/components/PlanetInfo/PlanetInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+interface PlanetFacts {
+  name: string;
+  id: number;
+  mass: number;
+  distance_from_sun: number;
+  length_of_year: number;
+  number_of_moons: number;
+  gravity: number;
+}
```
Comment (ConnorAndersonLarson) [inline src/components/PlanetInfo/PlanetInfo.tsx:12]: We can pull this from our interface file now.

**Review by ConnorAndersonLarson — APPROVED** [review]
Looks great, at some point we can utilize our Interface file for the planet interface (see single comment) but that can happen later! Also could probably remove some commented out code later.

File: src/components/App/App.tsx [hunk 5]
```diff
@@ -40,6 +39,7 @@ class App extends React.Component<{}, AllData> {
         <main>
           <SortBox updateSort={this.updateSort} />
           <Planetarium allPlanets={this.state.allPlanets} />
+          <PlanetInfo currentPlanet={this.state.allPlanets[2]} />
```
Comment (knees4bees) [inline src/components/App/App.tsx:42]: @alexthompson207 What's the job of the `[2]` here?

File: src/components/PlanetInfo/PlanetInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+          <li><b>Number of Moons: {number_of_moons}</b></li>
+          <li><b>Length of year: {length_of_year}</b></li>
+          <li><b>Gravity: {gravity}</b></li>
+        </ul>
+      </article>
+    </section>
+  )
+}
+
+export default PlanetInfo;
```
Comment (knees4bees) [inline src/components/PlanetInfo/PlanetInfo.tsx:39]: This looks great and is an awesome start!

**Review by knees4bees — APPROVED** [review]
Solid work, Alex!

**Pull request merged**